src/taxonomy/TaxonomyListPage.test.jsx

```diff
@@ -84,6 +88,54 @@ describe('<TaxonomyListPage />', async () => {
     });
   });
 
+  it('downloads the taxonomy template csv', async () => {
+    useIsTaxonomyListDataLoaded.mockReturnValue(true);
+    useTaxonomyListDataResponse.mockReturnValue({
+      results: [{
+        id: 1,
+        name: 'Taxonomy',
+        description: 'This is a description',
+      }],
+    });
+    const { getByTestId } = render(<RootWrapper />);
+    const importMenu = getByTestId('taxonomy-download-template');
+    expect(importMenu).toBeInTheDocument();
+    await act(async () => {
+      fireEvent.click(importMenu);
+    });
+
+    const importButton = getByTestId('taxonomy-download-template-csv');
+    expect(importButton).toBeInTheDocument();
+    await act(async () => {
+      fireEvent.click(importButton);
+    });
+    expect(getTaxonomyTemplateFile).toHaveBeenCalled();
+  });
+
+  it('downloads the taxonomy template json', async () => {
+    useIsTaxonomyListDataLoaded.mockReturnValue(true);
+    useTaxonomyListDataResponse.mockReturnValue({
+      results: [{
+        id: 1,
+        name: 'Taxonomy',
+        description: 'This is a description',
+      }],
+    });
+    const { getByTestId } = render(<RootWrapper />);
+    const importMenu = getByTestId('taxonomy-download-template');
+    expect(importMenu).toBeInTheDocument();
+    await act(async () => {
+      fireEvent.click(importMenu);
+    });
+
+    const importButton = getByTestId('taxonomy-download-template-json');
+    expect(importButton).toBeInTheDocument();
+    await act(async () => {
+      fireEvent.click(importButton);
+    });
+    expect(getTaxonomyTemplateFile).toHaveBeenCalled();
+  });
```

Comment:
This and the above test seem to be exactly the same with very minor differences, so we can use parametrisation here as well. 

```suggestion
it.each(['CSV', 'JSON'])('downloads the taxonomy template %s', async (fileFormat) => {
    useIsTaxonomyListDataLoaded.mockReturnValue(true);
    useTaxonomyListDataResponse.mockReturnValue({
      results: [{
        id: 1,
        name: 'Taxonomy',
        description: 'This is a description',
      }],
    });
    const { findByRole } = render(<RootWrapper />);
    const importMenu = await findByRole('button', {name: /download template/i});
    fireEvent.click(importMenu);
    const importButton = await findByRole('button', {name: `${fileFormat} template` });
    fireEvent.click(importButton);
    expect(getTaxonomyTemplateFile).toHaveBeenCalled();
  });
```

Note that I've simplified the test somewhat. 
- Using findByRole etc is preferred to finding by test id.
- No need for act around fireEvent
- No need to assert the presence of a component in the document, if the component is missing for some reason then the findByRole will still cause the test to fail.

Reply:
Thank you for the explanation @xitij2000!
Done 7f6319ea66e31115ffe650d2baa237ffa597bf42!
But I didn't manage to find a way to check that the download really started.